clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -232,12 +235,19 @@ opcode_mix_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         instr_init(dcontext_.dcontext, &instr);
         app_pc next_pc =
             decode_from_copy(dcontext_.dcontext, decode_pc, trace_pc, &instr);
-        if (next_pc == NULL || !instr_valid(&instr)) {
```

Comment:
Not sure if we should "fix" `instr_valid()` to work with DR_ISA_REGDEPS instructions (more overhead? unexpected behavior?) or if we should "fix" the analyzers instead like I did here.

Reply:
I think instr_valid() should return true for a decoded regdep instr.  OP_INVALID seems incorrect for any decoded instr.  So the regdep decoder should use OP_UNDECODED -- or will that trigger up-decode requests?  Maybe we can add logic to those requests.

Reply:
OP_UNDECODED seems to work. Done.